runtime/floats.c

```diff
@@ -523,7 +523,7 @@ CAMLprim value caml_nextafter_float(value x, value y)
   return caml_copy_double(caml_nextafter(Double_val(x), Double_val(y)));
 }
 
-#ifndef HAS_C99_FLOAT_OPS
+#ifndef HAS_WORKING_FMA
```

Comment:
Is that change correct? The way I read it, hosts with `HAS_C99_FLOAT_OPS` but not `HAS_WORKING_FMA` (that is, Cygwin) will get all the other-c99-ops below, such as `caml_frexp_float`, also defined (when they shouldn't, right?). I would expect a block that is *only* about FMA (and not other operations) to be under `#ifndef..#endif` brackets first, and a separate block for other float ops to be given separately.

(It looks like Cygwin could change the definition of some of the macro-stuff used in the FMA emulation (eg. the float literals) to real C99 operations instead of their emulation, so that the buildup code for FMA would depend on whether float-ops are there or not, but that makes thing more complex, probably for little gain.)

Reply:
`caml_frexp_float` is always defined. This change extends as far as the `#endif` on line 539 and is a set of definitions only used in the `#else` branch in `caml_fma`?

Reply:
Ah indeed, I misread. (It's the second time I misread your C code in a review...) Thanks!